spark/src/main/scala/org/apache/spark/sql/delta/v2/interop/README.md

```diff
@@ -0,0 +1,22 @@
+# V2 Connector Interop
+
+This package contains abstract traits that enable interoperability between the V1 and V2 Delta connectors.
+
+## Purpose
+
+These abstractions allow V1 utilities to be reused by the V2 connector through adapters.
+
+## Usage
+
+For reusing V1 connector code in V2:
+
+1. **Refactor V1 utilities** to depend on abstract traits (`AbstractMetadata`, `AbstractProtocol`, `AbstractCommitInfo`) instead of the concrete V1 implementations.
+
+2. **Implement adapters in V2 connector** that extend these abstractions, wrapping Kernel's action types (e.g., Kernel's `Metadata` → `AbstractMetadata`).
+
+## Traits
```

Comment:
please say these are examples, as this is not going to be exhaustive list. 
in fact, we can not have this list at all. all the descriptions are anyways present as class docs.